zipline/data/adjustments.py

```diff
@@ -283,13 +292,29 @@ def get_df_from_table(self, table_name, convert_dates=False):
             else {}
         )
 
-        return pd.read_sql(
+        result = pd.read_sql(
             'select * from "{}"'.format(table_name),
             self.conn,
             index_col='index',
             **kwargs
         ).rename_axis(None)
 
+        if not len(result):
```

Comment:
TIL `len(result)` is faster than `result.empty`. It seems `result.empty` performs extra checks we might not care about in this case.

Slightly faster, we could do `len(result.index)` instead.

Reply:
`empty` will be true if either rows or columns are empty. In this case, we only care about empty rows, so I think `len` is preferable (we never expect no columns, so they should have the same behavior here, but `len` expresses what we actually expect here).

re: `len(df.index)`, the implementation of `DataFrame.__len__` is just:

```
In [5]: pd.DataFrame.__len__??
Signature: pd.DataFrame.__len__(self)
Source:
    def __len__(self):
        """Returns length of info axis, but here we use the index """
        return len(self.index)
```

Adding an extra function call here isn't going to make a meaningful difference relative to the other things happening in this function, so I'd prefer the simpler expression here. If we were concerned with that level of micro performance here, much more important would be things like caching the column order produced by `keysorted`.